core/src/main/scala/internal/NettySupport.scala

```diff
@@ -224,15 +223,13 @@ private[http] object NettySupport {
           buffer = Chunk.concat(Seq(buffer, chunk.content.toChunk))
       }
 
-      // No more data available on the socket. If we have some buffered data
-      // enqueue it to the content queue and clear the buffer.
+      // No more data available on the socket. Enqueue the buffered
+      // data to the content queue and clear the buffer.
       override def channelReadComplete(ctx: ChannelHandlerContext) = {
-        if(buffer.nonEmpty) {
```

Comment:
This is because sometimes we get `channelReadComplete` for empty messages. Previously nothing was added to to content queue so we were in a deadlock situation because `channel.read()` is called as soon as a chunk is pulled from the content queue. Now when it happens we just push an empty chunk to the queue.